### PR TITLE
Replace all file dialogs with new FileDialog

### PR DESCRIPTION
**Pull request by stefan-kolb**
and set default FileFilter or aggregate FileFilter for all supported types.

- [x] Database from AUX
- [x] Remove BrowseAction

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Why do you kill the BrowseActio? It already uses the new FileDialogs and is used in many places (because eseentially the dialogs are everywhere the same)! It should be necessary to only call the setDefaultExtension method there. Or to inlcude it.

**Comment by stefan-kolb** [comment]
I just want to keep it simple and consistent. Either use Browse action everywhere or filedialog. I think for saving one or two lines its not worth keeping different classes here and there, but I'm open for suggestions.

**Comment by stefan-kolb** [comment]
Additionally, the BrowseAction does not make it obvious or easy to understand what it really does without inspecting the BrowseAction class. I'm open for suggestions and a different decision here. Maybe we should discuss this in a Devcall or other Devs can quickly take a look at the code. In my opinion the small code duplication makes it easier to understand whats happening here and therefore might be preferable over the encapsulation inside BrowseAction.

**Comment by Siedlerchr** [comment]
I gave the BroweAction a second thought and yes, it could be removed.

File: src/main/java/net/sf/jabref/gui/exporter/CustomExportDialog.java [hunk 1]
```diff
@@ -104,7 +104,12 @@ public CustomExportDialog(final JabRefFrame parent) {
         cancel.addActionListener(e -> dispose());
 
         JButton browse = new JButton(Localization.lang("Browse"));
-        browse.addActionListener(BrowseAction.buildForFile(layoutFile, FileExtensions.LAYOUT));
+        FileDialog dialog = new FileDialog(null).withExtension(FileExtensions.LAYOUT);
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/exporter/CustomExportDialog.java]: I think you should pass the jFrame parent here instead if null

File: src/main/java/net/sf/jabref/gui/protectedterms/NewProtectedTermsFileDialog.java [hunk 2]
```diff
@@ -49,7 +51,12 @@ public NewProtectedTermsFileDialog(JabRefFrame mainFrame, ProtectedTermsLoader l
 
     private void setupDialog() {
         JButton browse = new JButton(Localization.lang("Browse"));
-        browse.addActionListener(BrowseAction.buildForFile(newFile, FileExtensions.TERMS));
+        FileDialog dialog = new FileDialog(null).withExtension(FileExtensions.TERMS);
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/protectedterms/NewProtectedTermsFileDialog.java]: same as above?

**Comment by stefan-kolb** [comment]
What we should discuss here is the state model of the FileDialog class. All methods alter the state of the internal JFileChooser dialog. But if we show a Directory selection dialog, we cannot revert back to a File selection dialog. Also if we use multiple files selection this is used in former single file selection dialogs...

**Comment by Siedlerchr** [comment]
Regarding the state: I do not think that this causes a big problem here, as we create a new Object of the FileDialog including the underlying internal jFileChooser for each dialog where it is used.

The other point: Directories only is clear. And MultipleFiles vs Single Files should be clear, too. However, even in a muliple file dialog I am able to select only file.

**Comment by stefan-kolb** [comment]
@Siedlerchr Yeah, but if you don't know this - it creates potential bugs. because not all state transitions are allowed as I explained.

**Comment by Siedlerchr** [comment]
I know what you mean. You could introduce a` private boolean called = false` and check for it and throw an exception if it's true on entering any other method...

**Comment by grimes2** [comment]
This state problem is also present in current master. So for example in database properties the encoding is set to the first entry Big5.

**Comment by Siedlerchr** [comment]
@stefan-kolb  Take a look at #1803,  seems like the full path (including filename) is stored in the prefs.

**Comment by stefan-kolb** [comment]
Is this a new One caused by this pr or was this the case before?

**Comment by stefan-kolb** [comment]
@koppor @grimes2 
> This state problem is also present in current master. So for example in database properties the encoding is set to the first entry Big5.

I don't get this one. What doo you mean?

**Comment by grimes2** [comment]
Sorry, my fault. I've no evidence, that the bug is caused also by this pull request. I've opened a new issue for this: #1806

**Pull request merged**